libs/api/ebpf_api.cpp

```diff
@@ -1948,6 +2144,267 @@ ebpf_program_unload(_In_ struct bpf_program* program)
     return EBPF_SUCCESS;
 }
 
+static ebpf_result_t
+_load_native_module(
+    _In_ const std::wstring& service_path,
+    _In_ const GUID* module_id,
+    _Out_ size_t* count_of_maps,
+    _Out_ size_t* count_of_programs)
+{
+    ebpf_result_t result = EBPF_SUCCESS;
+    uint32_t error = ERROR_SUCCESS;
+    ebpf_protocol_buffer_t request_buffer;
+    ebpf_operation_load_native_module_request_t* request;
+    ebpf_operation_load_native_module_reply_t reply;
+    size_t service_path_size = service_path.size() * 2;
+
+    *count_of_maps = 0;
+    *count_of_programs = 0;
+
+    size_t buffer_size = offsetof(ebpf_operation_load_native_module_request_t, data) + service_path_size;
+    request_buffer.resize(buffer_size);
+
+    request = reinterpret_cast<ebpf_operation_load_native_module_request_t*>(request_buffer.data());
+    request->header.id = EBPF_OPERATION_LOAD_NATIVE_MODULE;
+    request->header.length = static_cast<uint16_t>(request_buffer.size());
+    request->module_id = *module_id;
+    memcpy(
+        request_buffer.data() + offsetof(ebpf_operation_load_native_module_request_t, data),
+        (char*)service_path.c_str(),
+        service_path_size);
+
+    error = invoke_ioctl(request_buffer, reply);
+    if (error != ERROR_SUCCESS) {
+        result = win32_error_code_to_ebpf_result(error);
+        goto Done;
+    }
+
+    ebpf_assert(reply.header.id == ebpf_operation_id_t::EBPF_OPERATION_LOAD_NATIVE_MODULE);
+    *count_of_maps = reply.count_of_maps;
+    *count_of_programs = reply.count_of_programs;
+
+Done:
+    return result;
+}
+
+static ebpf_result_t
+_load_native_programs(
+    _In_ const GUID* module_id,
+    _In_opt_ const ebpf_program_type_t* program_type,
+    size_t count_of_maps,
+    _Out_writes_(count_of_maps) ebpf_handle_t* map_handles,
+    size_t count_of_programs,
+    _Out_writes_(count_of_programs) ebpf_handle_t* program_handles)
+{
+    ebpf_result_t result = EBPF_SUCCESS;
```

Comment:
please add EBPF_LOG_* macros for tracing.

Reply:
Done.